src/foundry/common/documents/setting.mjs.d.ts

```diff
@@ -0,0 +1,85 @@
+// FOUNDRY_VERSION: 10.291
+
+import Document, { DocumentMetadata } from "../abstract/document.mjs";
+import * as fields from "../data/fields.mjs";
+
+declare global {
+  type SettingData = BaseSetting.Properties;
+}
+
+/**
+ * The Document definition for a Setting.
+ * Defines the DataSchema and common behaviors for a Setting which are shared between both client and server.
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface BaseSetting extends BaseSetting.Properties {}
+declare class BaseSetting extends Document<BaseSetting.SchemaField, BaseSetting.Metadata> {
+  /**
+   * @param data    - Initial data from which to construct the Setting
+   * @param context - Construction context options
+   */
+  constructor(data: BaseSetting.ConstructorData, context?: DocumentConstructionContext);
+
+  static override metadata: Readonly<BaseSetting.Metadata>;
+
+  static override defineSchema(): BaseSetting.Schema;
+}
+export default BaseSetting;
+
+declare namespace BaseSetting {
+  type Metadata = Merge<
+    DocumentMetadata,
+    {
+      name: "Setting";
+      collection: "settings";
+      label: "DOCUMENT.Setting";
+      labelPlural: "DOCUMENT.Settings";
+      permissions: {
+        create: "SETTINGS_MODIFY";
+        update: "SETTINGS_MODIFY";
+        delete: "SETTINGS_MODIFY";
+      };
+    }
+  >;
+
+  type SchemaField = fields.SchemaField<Schema>;
+  type ConstructorData = UpdateData;
```

Comment:
As with `BaseMacro`, this should be updated. At least `key` and `value` are required.

Reply:
Updated